Documentation/ceph-csi-drivers.md

```diff
@@ -124,5 +128,50 @@ when pod get spawned and destroyed at time of pod delete.
 Refer to [ephemeral-doc](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes) for more info.
 Also, See the example manifests for an [RBD ephemeral volume](https://github.com/rook/rook/tree/{{ branchName }}/deploy/examples/csi/rbd/pod-ephemeral.yaml) and a [CephFS ephemeral volume](https://github.com/rook/rook/tree/{{ branchName }}/deploy/examples/csi/cephfs/pod-ephemeral.yaml).
 
-### Prerequisites
-Kubernetes version 1.21 or greater is required.
+## CSI-Addons Controller
+
+The CSI-Addons Controller handles the requests from users to initiate an operation. Users create a CR that the controller inspects, and forwards a request to one or more CSI-Addons side-cars for execution.
+
+### Deploying the controller
+
+Users can deploy the controller by running the following commands:
+
+```bash
+kubectl create -f https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/v0.3.0/deploy/controller/crds.yaml
+kubectl create -f https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/v0.3.0/deploy/controller/rbac.yaml
+kubectl create -f https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/v0.3.0/deploy/controller/setup-controller.yaml
+```
+
+This creates the required crds and configure permissions.
+
+### Enable the CSI-Addons Sidecar
+
+To use the features provided by the CSI-Addons, the `csi-addons`
+containers need to be deployed in the RBD provisioner and nodeplugin pods,
+which are not enabled by default.
+
+Execute the following command in the cluster to enable the CSI-Addons
+sidecar:
+
+* Update the `rook-ceph-operator-config` configmap and patch the
+ following configurations
+
+```bash
+kubectl patch cm rook-ceph-operator-config -nrook-ceph -p $'data:\n "CSI_ENABLE_CSIADDONS": "true"'
+```
+
+* After updating the configmap with those settings, a new sidecar
+ should now start automatically in the CSI provisioner and nodeplugin pod.
+
+> NOTE: Make sure the version of ceph-csi used is v3.5.0+
+
+### CSI-ADDONS Operation
+
+CSI-Addons supports the following operations:
+
+- ReclaimSpace
+  - [Creating ReclaimSpaceJob](https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/reclaimspace.md#reclaimspacejob)
+  - [Creating ReclaimSpaceCronJob](https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/reclaimspace.md#reclaimspacecronjob)
+  - [Annotating PersistentVolumeClaims](https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/reclaimspace.md#annotating-perstentvolumeclaims)
+- NetworkFence
```

Comment:
```suggestion
- Network Fencing
```

Reply:
NetworkFence is a kind so I guess it is fine this way.